arrow/src/array/array.rs

```diff
@@ -198,10 +198,14 @@ pub trait Array: fmt::Debug + Send + Sync + JsonEqual {
     }
 
     /// Returns the total number of bytes of memory occupied by the buffers owned by this array.
```

Comment:
```suggestion
    /// Returns the total number of bytes of memory pointed to by this array.
    /// The buffers store bytes in the Arrow memory format, and include the data as well as the validity map.
```

The distinction between `buffers` and `physically occupied` has always been somewhat confusing to me. Perhaps we can take this opportunity to clarify what they mean